### DISCUSSION TITLE
Making Snow / Fireworks plugin

### DISCUSSION THREAD

**Discussion by PH5HP** [e107inc/e107#5374] [Ideas]
Hi,

I found something interesting, but I am not great in turning something like this into a plugin.

https://www.cssscript.com/christmas-snowfall-new-years-fireworks/

The URL above describes a plugin for sown and/or fireworks on your website. Anyone who can help me turning this into a plugin or working together. Would be nice.

Kind regards,
Henk - www.ph5hp.nl

**Comment by rica-carv** [comment]
Real ankward site. It doesn't return to the previous page, it stick to itself....
I simply don't trust those sites....

Anyway, if you really want to use it, it's just a simple js library, so you can load it via regular javascript loading...

**Reply by PH5HP** [reply]
Hi,

My mistake :-)

https://www.cssscript.com/tag/snow/

More snowscripts. It is just www.cssscript.com (main url). Seems to me that it is no problem website. I was thinking to embed it in the them or something, but making a plugin is better, because more people could use it.

There is a snow plugin, but the snow animation is outdated and limited to 1024 pixels. So I was looking at some more slick alternatives..

Kind regards,
Henk - www.ph5hp.nl

**Reply by Alex-e107nl** [reply]
hahahahah look at [https://codepen.io/tag/snow](https://codepen.io/tag/snow) and this one.. [https://codepen.io/nodws/pen/dPbWOze](https://codepen.io/nodws/pen/dPbWOze) only css... relaxed snow... place:
```
div class="cssnow"></div>
<script>for(let i=0;i<9;i++){
  document.querySelector('.cssnow').innerHTML += '<span>';}</script>
```
in your theme.html and this css in your style.css:

```
.cssnow {
   pointer-events: none;
   position: absolute;
}

.cssnow span {
   display: block;
   background: #fff;
   border-radius: 50%;
   width: 9px;
   height: 12px;
   box-shadow: -100px -290px 0px -2px #fff5;
 rotate: 35deg;
  filter:blur(3px)
}

@keyframes cssnowflakes-fall {
   0% {
      top: -10%;
   }

   100% {
      top: 100%;
   }
}

@keyframes cssnowflakes-shake {
   0% {
      transform: translateX(0px);
   }

   50% {
      transform: translateX(80px);
   }

   100% {
      transform: translateX(0px);
   }
}

.cssnow span {
   position: fixed;
   top: -10%;
   z-index: 9999;
   animation-name: cssnowflakes-fall, cssnowflakes-shake;
   animation-duration: 10s, 3s;
   animation-timing-function: linear, ease-in-out;
   animation-iteration-count: infinite, infinite;
   animation-play-state: running, running;
}

.cssnow span:nth-of-type(0) {
   left: 1%;
   animation-delay: 0s, 0s;
}

.cssnow span:nth-of-type(1) {
   left: 10%;
   animation-delay: 1s, 1s;
  scale:2
}

.cssnow span:nth-of-type(2) {
   left: 20%;
   animation-delay: 6s, 0.5s;
}

.cssnow span:nth-of-type(3) {
   left: 30%;
   animation-delay: 4s, 2s;
}

.cssnow span:nth-of-type(4) {
   left: 40%;
   animation-delay: 2s, 2s;
   
}

.cssnow span:nth-of-type(5) {
   left: 50%;
   animation-delay: 8s, 3s;scale:3
}

.cssnow span:nth-of-type(6) {
   left: 60%;
   animation-delay: 6s, 2s;scale:3;
  filter:blur(5px)
}

.cssnow span:nth-of-type(7) {
   left: 70%;
   animation-delay: 2.5s, 1s;
}

.cssnow span:nth-of-type(8) {
   left: 80%;
   animation-delay: 1s, 0s;
}

.cssnow span:nth-of-type(9) {
   left: 90%;
   animation-delay: 3s, 1.5s;
}
```
Result is here [https://www.fysiowebdesign.nl/demo03/news](https://www.fysiowebdesign.nl/demo03)

**Reply by PH5HP** [reply]
Hi,

In theme.html ? Or in theme.php? And if so: where in theme.php?...

**Reply by LaocheXe** [reply]
> Hi,
> 
> In theme.html ? Or in theme.php? And if so: where in theme.php?...

I got it working on my [Falling Holidays eXeSystem](https://github.com/LaocheXe/Falling-Holidays-eXeSystem) plugin, I will update the git, all you have to do is enable the snow, select CSSnow, and add the CSSnow Menu and should work.

**Reply by PH5HP** [reply]
Hi, 

I was just at it when you post the message above this one.

Some remarks. I am using a greater resolution then 1920x1080. So the snow does not fall from the whole width of my screen and also not to the bottom of my webpage (this maybe configurable).  The same for the Chrismas lights.

Also the snow is falling behind my main colom of my website and not over it .

Kind regards,
Henk - www.ph5hp.nl (if you will take a look)

**Reply by PH5HP** [reply]
Also it is not falling from the top of the screen ... For example just under the main menu.

**Reply by PH5HP** [reply]
Another thing I discovered is that if the snow is active some links are no links anymore and therefore not clickable.

**Reply by LaocheXe** [reply]
Did you enable the menu in menu manager?

> Also it is not falling from the top of the screen ... For example just under the main menu.

**Reply by PH5HP** [reply]
Yes, I did ...

**Reply by PH5HP** [reply]
Maybe a resolution issue?... I am using 2560px x 1440px ...

**Reply by Alex-e107nl** [reply]
Well the plugin works perfectly with the snow css THX @LaocheXe ! And i have a 2560x1440 screen.

**Reply by PH5HP** [reply]
Hi Alex,

I activated as a menu. Did you too and if so does the menu position matter? Or did you add it to your theme? If so: how?

Kind regards,
Henk - www.ph5hp.nl

**Reply by Alex-e107nl** [reply]
Yep i just choose a menu, there are more from the plugin to choose from but i used the CSSnow on the HOME Page, you can look in the source code of your page if it is loading the right css...

`<link  rel="stylesheet" media="all" property="stylesheet" type="text/css" href="[/demo03/e107_plugins/fallingholidays/css/CSSnow.css?1735255559](view-source:https://www.fysiowebdesign.nl/demo03/e107_plugins/fallingholidays/css/CSSnow.css?1735255559)" /> `

And where you placed the menu you have to see:

```
 <div class="cssnow"></div>
			<script>for(let i=0;i<9;i++){
				document.querySelector(".cssnow").innerHTML += "<span>";}</script>
```

And it must work when you see those 2 :-)  
I am using a standard BS theme with 5.3.3 nothing special...

Also now working on [https://e107.nl/](https://e107.nl/) without any problems!

**Comment by LaocheXe** [comment]
> I found something interesting, but I am not great in turning something like this into a plugin.
> There is a snow plugin, but the snow animation is outdated and limited to 1024 pixels. So I was looking at some more slick alternatives..

Which plugin, it wouldn't have been this one: [Falling Holidays eXeSystem](https://github.com/LaocheXe/Falling-Holidays-eXeSystem)

I will be working on adding newer stuff to it.

**Reply by Jimmi08** [reply]
My users prefer only lights, still working in PHP 8.1 
![image](https://github.com/user-attachments/assets/dbe2ab74-7c7b-491b-af2f-3c6e558b3cb9)

**Reply by rica-carv** [reply]
Personally i think its a overkill to have a plugin just to put js on the theme.
I like to "control" js on my theme...
Maybe im paranoic, but...

Don't get me wrong, using a plugin, a module or insert js directly in theme its everybody choice... 

Its just my personal opinion.

**Comment by PH5HP** [comment]
Hi,

I like this one too.

https://codepen.io/redstapler/pen/MZXJZx

Kind regards,
Henk - www.ph5hp.nl

**Comment by PH5HP** [comment]
Hi,

If I activate this plugin and put it in the menu section at the top then the snow comes from that position and not the whole width of the screen. Just the left side remains without the snow. The right side is ok. This happens only if you set the SNOW TYPE on 3D SNOW. Also the links from the POPULAR MENU are not present anymore. See the screenshot.

When you set SNOW TYPE to SNOW STORM all is working OK.

Kind regards,
Henk - www.ph5hp.nl

![image](https://github.com/user-attachments/assets/823b5476-8211-4130-b617-39b46e22da31)

**Comment by tgtje** [comment]
Ahh old stuff, reminded me of a plug i adjusted (2015) probably used stuff from another thosedays, tested iff it still worked (php  8.2 local)
forgotten it reacts on mouse

https://github.com/user-attachments/assets/66eda0fe-8214-4af0-afe1-b4c12977742f
best wishes for 2025

**Comment by rica-carv** [comment]
You can also [use css](https://devsnap.me/css-snow-effects)...

**Reply by Alex-e107nl** [reply]
Hi @rica-carv That's what LaocheXe showed, it was really very simple to add a css snow thing to his plugin, if you only use that, very little happens with javascript on your site, 2 lines but otherwise very simple, it was nice that it went so quickly!

**Reply by LaocheXe** [reply]
Yeah, don't even need to edit anything code related, just install the plugin, go configure it (activate what you want, select the effects), and some need a menu enabled to work on the site. Other then that, it should work.

**Reply by rica-carv** [reply]
And besides, js is something you can turnoff on your browser, css no. 😉

**Reply by Jimmi08** [reply]
Plugin or theme?  Plugin, so it is the universal solution for any theme.  You switch the theme and it will still work. This is not question anymore. 

Custom code or somebody else plugin? It depends on your time and skills.  Initially, I copied that plugin and stripped everything I didn't need.  Because of the Nuke part of my site,  I had manually implemented the script into the Nuke theme too. So I understand @rica-carv's point. After Christmas, I had to remove it manually. BUT.  After years (and short of time) only e107's part uses Lights now because I can switch it On/Off easily.   I haven't time and energy each year to remember what to change and to uncomment to have it working manually in theme. 

Am I glad now that I have my copy?  Yes and no. Yes, because it still works and my site is safe from somebody else changes. No, I need to maintenance it.  

Why I haven't use @LaocheXe plugin directly?  Because name.  Yes, it is too complicated.  It is only reason.

**Reply by tgtje** [reply]
Ah yes css vs js. but the files grow in code length, of course better imaging result need more.
but plugin seems easier click on, click off. 
anyway the way i did it using things ( by name it seems @LaocheXe used it too, or in 2015 made first plugin ( we used a lot of stuff for tinkering from others)???
i solved it (then)  with 1 image 2 js files 1 xml (plug) 1 php file. (as simple as possible)
For those interested [repo](https://github.com/tgtje/xmaslights/tree/main/xmaslights).
I still remember we had plugins that put christmas wishes card form ( set on date) to a site. Ahh my mind..... Alf or Fizi ???

**Reply by rica-carv** [reply]
> After Christmas, I had to remove it manually.

@Jimmi08 You could use timers, i've used them in my theme in old e107, and it worked like a charm, i didn't had to turn it on & off manually. Just set a time period where it would be on, and that's it. That's the beauty of PHP.... 😄 

> but plugin seems easier click on, click off.

@tgtje My personal view, once more. If it's for admin control of the plugin, on the backend, for me, i could use a plugin, but setup a menu on the frontend and left the end user control the "script", it's a overkill, for me.
On the limit, for me, a simple on/off button would do...

Right now, we are on the IA era, and for instance, we can control if a user wants or doesn't want something displayed without the intervention of the user.
For instance, i can make the e107 welcome message pop x times on first users visits and then never more... Even i can put the message on or off just by knowing if a user is already logged in or not...

**Reply by tgtje** [reply]
@rica-carv i understand, no problem. In regards to AI, thats another discussion worth. (later point in time will surely rise).
